tests/state/ViewerState/dataset_slice.test.ts

```diff
@@ -9,58 +9,54 @@ import {
   MOCK_DATASET_WITHOUT_BACKDROP,
   MockFeatureKeys,
 } from "./constants";
+import { setDatasetAsync } from "./utils";
```

Comment:
Added this utility method in this and the other tests. This is because the listener for the dataset setter in `time_slice` calls an async function that sets the state, and react-testing-library will output a bunch of warnings about asynchronous state updates. The helper function manages this for Dataset.